src/internals.jl

```diff
@@ -135,34 +143,46 @@ end
 # check_x0!, check_lvar!, check_uvar!
 # Functions to check user given arrays. 
 ####################################################
+function check_x0!(x0_::AbstractVector, lvar::AbstractVector, uvar::AbstractVector, x0, cache::VecchiaCache)
+    if isnothing(x0) 
+        return
+    end
+    
+    v = view(x0_, cache.diagL)
+    uvar_diag = view(uvar, cache.diagL)
+    lvar_diag = view(lvar, cache.diagL)
+    
+    # clamp x0_ lvar and uvar 
+    if mapreduce(x -> x > 0, &, view(x0, cache.diagL))       
+        view(x0_, 1:cache.nnzL) .= x0
```

Comment:
Ypu forgot to keep the initial value of `z`!
I said to be careful for that on Slack ;)
```
view(x0_, (1:cache.n).+cache.nnzL) .= log.(view(iVecchiaMLE.x0, cache.diagL))
```